### DISCUSSION TITLE
2FA prompts lost with ssh command

### DISCUSSION THREAD

**Discussion by tferrin** [blinksh/blink#1568] [Support]
I'm a relatively new blink user, but I've read the docs and looked through Discussions and Issues and haven't found anything similar to this problem. But before I create a new Issue, I wanted to see if anyone else has come across this situation.

I ssh to a number of servers that require 2FA as part of the authentication process. Typically a session with one of these hosts goes something like this when using ssh from a linux or mac client:

```
MyMac> ssh plato
(tef@plato.xxxx.edu) Duo two-factor login for tef
Enter a passcode or select one of the following options:

 1. Duo Push to XXX-XXX-7592
 2. Phone call to XXX-XXX-7592
 3. SMS passcodes to XXX-XXX-7592

Passcode or option (1-3):  1
(tef@plato.xxxx.edu) Success. Logging you in...
Remember connection authentication from 99.xxx.yyy.zzz for 12 hours? [y/N] y
Plato>
``` 
(The "1" and "y" were input provided by me during the 2FA exchange.)

My problem with blink is that I usually only see a few words of the above output. The authentication process pauses after the initial ssh key exchange, but then I just have to know I need to first type a "1", then go to the Duo app on my phone and click Accept, then type a "y" if I want the authentication to be remembered for the next 12 hours.

I've tried ssh -vvv and get the normal debugging output, often with a few words from the above text sprinkled in, but that's as good as it gets. Am I missing something obvious? It seems like blink is not expecting to receive text back as part of the authentication process, or if it is then there's a race condition that prevents the full text from being received and displayed.

**Comment by carloscabanero** [comment]
This is a bug, will fix, tracking on https://github.com/blinksh/blink/issues/1577
